core/src/dird/bsr.cc

```diff
@@ -424,20 +427,19 @@ static uint32_t write_bsr_item(RestoreBootstrapRecord* bsr,
                  edit_uint64(bsr->VolParams[i].StartAddr, ed1),
                  edit_uint64(bsr->VolParams[i].EndAddr, ed2));
 
-    count = write_findex(bsr->fi.get(), bsr->VolParams[i].FirstIndex,
-                         bsr->VolParams[i].LastIndex, buffer);
+    int32_t start = bsr->VolParams[i].FirstIndex;
+    int32_t end = bsr->VolParams[i].LastIndex;
+    count = write_findex(bsr->fi.get(), start, end, buffer);
```

Comment:
`bsr->VolParams[i].First/LastIndex` are `uint32_t`, so `start` and `end` should be `uint32_t` too, as well as `write_findex` parameters and all subsequent usages under that.

Reply:
I think the indices are signed because `RestoreBootstrapRecordFileIndex` uses signed integers to store the file indices to restore.  Ill look into why that is. Otherwise you are correct!

Reply:
Seems like file indices are stored as 64bit signed integers in the database and then casted to signed 32bit.  I added ASSERTs to ensure that the indices are >= 0 and cast them to `uint32_t`.